api/lib/opentelemetry/distributed_context/propagation/trace_parent.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+module OpenTelemetry
+  module DistributedContext
+    module Propagation
+      # A TraceParent is an implemenation of the W3C trace context specification
+      # {SpanContext}
+      class TraceParent
+        InvalidFormatError = Class.new(Error)
+        InvalidVersionError = Class.new(Error)
+        InvalidTraceIDError = Class.new(Error)
+        InvalidSpanIDError = Class.new(Error)
+
+        TRACE_PARENT_HEADER = 'traceparent'
+        SUPPORTED_VERSION = 0
+        private_constant :SUPPORTED_VERSION
+        MAX_VERSION = 254
+        private_constant :MAX_VERSION
+
+        REGEXP = /^([A-Fa-f0-9]{2})-([A-Fa-f0-9]{32})-([A-Fa-f0-9]{16})-([A-Fa-f0-9]{2})(-.*)?$/.freeze
```

Comment:
Would this (and the parsing) be more clear if we used named captures?

Reply:
sure it's easy to add.